### PR TITLE
ready for more providers

### PR DESCRIPTION
**Pull request by Amoki**

### CODE REVIEW

File: player/views/remote/library.py [hunk 1]
```diff
@@ -6,21 +6,20 @@
...
-            musics_searched = youtube.search(ids=[regexVideoId.search(request.POST.get('query')).group(1),])
-        template_library = render_to_string("include/remote/library.html", {"musics": musics_searched, "tab": "youtube-list-music"})
+    if request.is_ajax() and request.session.get('room', False) and request.POST.get('source'):
+        source = Source.objects.get(name=request.POST.get('source'))
+
+        musics_searched = source.search(query=request.POST.get('query'))
+
+        template_library = render_to_string("include/remote/library.html", {
+            "musics": musics_searched,
+            "tab": source.name.lower() + "-list-music",
```
Comment (Amoki) [inline player/views/remote/library.py]: can't be in front ?

File: player/views/remote/remote.py [hunk 2]
```diff
@@ -70,7 +70,11 @@ def update_remote(request):
             return HttpResponse("Error while refreshing the library, please reload the page", status=409)
 
         player_updated = json.loads(render_remote(room))
-        player_updated['template_library'] = render_to_string("include/remote/library.html", {"musics": musics, "tab": "library-list-music", "more_musics": more_musics})
+        player_updated['template_library'] = render_to_string("include/remote/library.html", {
+            "musics": musics,
+            "tab": "library-list-music",
```
Comment (Amoki) [inline player/views/remote/remote.py]: same

File: source_modules/youtube/models.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+import random
+import string
+import re
+
+from apiclient.discovery import build as youtube_api
+
+from amoki_music.settings.common import YOUTUBE_KEY
```
Comment (Amoki) [inline source_modules/youtube/models.py]: from django.conf import settings

File: static/css/style.less [hunk 4]
```diff
@@ -305,6 +311,14 @@ button > .glyphicon, a > .glyphicon {
 	}
 }
 
+.soundcloud-list-music {
```
Comment (Amoki) [inline static/css/style.less]: Can we have a multi-source css instead a copy/paste for each source :/

Comment (Eirika) [inline static/css/style.less]: .youtube-list-music .soundcloud-list-music { ... }

Comment (Eirika) [inline static/css/style.less]: [class$="-list-music"] { ... }

File: static/js/remote/ajax.js [hunk 5]
```diff
@@ -40,42 +40,40 @@ $(document).on('submit', '.ajax-next, .ajax-dead-link', function(e) {
...
-      $("#youtube").addClass("active");
-      $("#tab_btn_library").removeClass("active");
-      $("#library").removeClass("active");
-      $("#list-youtube").slideUp();
-      $("#list-youtube").promise().done(function() {
-        $(".youtube-list-music").remove();
-        $("#list-youtube").html(data.template_library);
-        $("#list-youtube").slideDown();
-        $("#list-youtube").promise().done(function() {
-          $("#btn-search").children("i").attr("class", "fa fa-youtube-play");
```
Comment (Amoki) [inline static/js/remote/ajax.js]: ça

File: static/js/remote/ajax.js [hunk 6]
```diff
@@ -40,42 +40,40 @@ $(document).on('submit', '.ajax-next, .ajax-dead-link', function(e) {
...
-          $("#btn-search").removeAttr('disabled');
```
Comment (Amoki) [inline static/js/remote/ajax.js]: et ça

**Pull request merged**